posthog/models/user.py

```diff
@@ -245,6 +248,14 @@ def leave(self, *, organization: Organization) -> None:
                     None if self.current_organization is None else self.current_organization.teams.first()
                 )
                 self.save()
+        self.update_billing_distinct_ids(organization)
+
+    def update_billing_distinct_ids(self, organization: Organization) -> None:
+        from ee.billing.billing_manager import BillingManager  # avoid circular import
+
+        if is_cloud():
+            license = License.objects.first_valid()
+            BillingManager(license).update_billing_distinct_ids(organization)
```

Comment:
Not strictly related to this PR, but it seems really inefficient to select the Cloud license from the database each time, when – correct me if I'm wrong – it's never going to change within the lifetime of a PostHog deployment (less than a couple days)

Reply:
Good idea